packages/embed-jsx/src/lib/EmbedJsx.ts

```diff
@@ -0,0 +1,115 @@
+import { MessageEmbed, ColorResolvable } from 'discord.js';
+
+const EMBED_TYPES = ['embed', 'title', 'field', 'timestamp', 'footer', 'description', 'url', 'image'] as const;
+
+type EmbedInformation =
+	| TitleInformation
+	| FieldInformation
+	| TimestampInformation
+	| FooterInformation
+	| DescriptionInformation
+	| UrlInformation
+	| ImageInformation;
+
+type EmbedData = TitleData | FieldData | TimestampData | FooterData | DescriptionData | UrlData | ImageData;
+
+const enum TsxTypes {
+	Title,
+	Description,
+	Url,
+	Field,
+	Timestamp,
+	Footer,
+	Image
+}
+
+// Received types
+type TitleInformation = [null, string];
+type DescriptionInformation = [null, string];
+type UrlInformation = [null, string];
+type FieldInformation = [{ title?: string; inline?: boolean }, string];
+type TimestampInformation = [null, number | string | Date | null];
+type FooterInformation = [null | { iconURL?: string }, string];
+type EmbedInitialInformation = [{ color: ColorResolvable }, ...EmbedData[]];
+type ImageInformation = [{ url: string }, null];
+
+// Returned types
+type TitleData = [TsxTypes.Title, string];
+type DescriptionData = [TsxTypes.Description, string];
+type UrlData = [TsxTypes.Url, string];
+type FieldData = [TsxTypes.Field, string, string, boolean];
+type TimestampData = [TsxTypes.Timestamp, Date];
+type FooterData = [TsxTypes.Footer, string, string?];
+type ImageData = [TsxTypes.Image, string];
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace EmbedJsx {
```

Comment:
Can you add typedoc annotations please?

Reply:
Docs will be done once i can finish up the readme